app/database_etl/location_utils/location_util_functions.py

```diff
@@ -124,28 +124,53 @@ def row_in_feature_layer(row: pd.Series, feature_layer: FeatureLayer) -> bool:
 # Get pin latitude and longitude for a record
 # Param geo_dfs = dictionary of city, state, and country DFs
 def get_record_coordinates(record: pd.Series, geo_dfs: dict) -> Tuple:
-    pin_regions = [record['country']]
-    pin_region_type = 'country' if record['country'] is not None else ''
-
-    for region_type in ['state', 'city']:
-        if record[region_type] and len(record[region_type]) > 0:
-            pin_regions = record[region_type]
-            pin_region_type = region_type
-            # once we've found more than 1 region for a given type
-            # we cannot compute locations from a more specific region type
-            # e.g. if we have multiple states, we cannot match each city with a state
-            # thus we must stop our selection here
-            if len(pin_regions) > 1:
-                break
-
-    if pin_region_type == '':
-        return None, None
-
-    # get latitude and longitude for the record
-    geo_df = geo_dfs[pin_region_type]
-    col_name = f"{pin_region_type}_name"
-    pin_lat = mean([geo_df[geo_df[col_name] == region_name].iloc[0]['latitude'] for region_name in pin_regions])
-    pin_lng = mean([geo_df[geo_df[col_name] == region_name].iloc[0]['longitude'] for region_name in pin_regions])
+    pin_lat, pin_lng = None, None
+
+    # First try to query a matching country for the record and temporarily assign
+    # its pin coordinates to it
+    if not record['country']:
+        # If there is no country associated with the record there is no way we can query a latlng, return None
+        return pin_lat, pin_lng
+    country_df = geo_dfs['country']
+    matching_country = country_df[country_df['country_name'] == record['country']].iloc[0]
+    pin_lat, pin_lng = matching_country['latitude'], matching_country['longitude']
```

Comment:
what happens if matching_country is none here? Is that ever the case, or are we sure that our country db is comprehensive?

Reply:
The country_df is constructed by taking the unique set of countries that exist in our data, matching_country should always return a record

Reply:
We do have cases where matching_country will be incomplete tho (e.g. no ISO2 code for vatican city - which we need to update when we move to airtable)

Reply:
got it!